### ISSUE TITLE
Improve installation instructions

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#1635]
I tried to follow the instructions at https://github.com/googlefonts/fontbakery#mac-os-x and found that 

- [ ] fontforge brew script no longer has a `--HEAD` option
- [ ] fontbakery's pypi package is very old and needs to be updated
- [ ] when I installed with `pip install .` then I got no executables in my PATH, so I wasn't able to run anythign

Would be great to fix this before 0.3.3

**Comment by felipesanches** [comment]
At some point I saw something like 0.2.0 on your terminal during our video call.
Does it mean MacOS users do not get the latest pypi packages that I've shipped for 0.3.1 and 0.3.2 ?

**Comment by felipesanches** [comment]
I would appreciate some help from @m4rc1e here as I presumed that fontbakery was running properly on Mac since I get constant feedback from him on pretty up-to-date versions of the program. So he has clearly got a good setup on Mac.

**Comment by felipesanches** [comment]
I don't have a computer running MacOS to test this, so I gotta collaborate with Mac users in order to get this right.

**Comment by felipesanches** [comment]
I did review the install instructions today and tested them on GNU+Linux. Looks good. I did not yet test the MacOS instructions. Anyone here with a mac would like to take a look?

To do that properly I'd suggest creating a fresh new virtualenv and activating it:

```
virtualenv venv
. venv/bin/activate
```

And then install the package from PyPI with:

```
pip install fontbakery
```

And try to use it. Some tests would still require installing things like fontforge, ots-sanitize, MSFVal, etc... which are already described on the README.md of this repo. But the above should let us know if there's anything seriously broken on Mac in case even this basic/minimal install fails.

**Comment by davelab6** [comment]
No dice:

```
20:45 $ mkdir tmp-fb-test-install
20:45 $ cd !$
20:45 $ virtualenv venv
Using base prefix '/usr/local/Cellar/python3/3.6.0_1/Frameworks/Python.framework/Versions/3.6'
New python executable in /Users/dcrossland/tmp-fb-test-install/venv/bin/python3.6
Also creating executable in /Users/dcrossland/tmp-fb-test-install/venv/bin/python
Installing setuptools, pip, wheel...done.
20:45 $ . venv/bin/activate
20:45 $ pip install fontbakery
Collecting fontbakery
  Downloading fontbakery-0.0.13.tar.gz (163kB)
    100% |████████████████████████████████| 163kB 3.6MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "/private/var/folders/tk/9btsy9sn7ybgnhksp880wm54003r41/T/pip-build-khslhlkr/fontbakery/setup.py", line 25, in <module>
        import fontforge
    ModuleNotFoundError: No module named 'fontforge'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/tk/9btsy9sn7ybgnhksp880wm54003r41/T/pip-build-khslhlkr/fontbakery/setup.py", line 27, in <module>
        raise Exception(msg)
    Exception: Python module `fontforge` is required. Install it with `apt-get install python-fontforge` or `brew install python; brew install fontforge --HEAD`
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/tk/9btsy9sn7ybgnhksp880wm54003r41/T/pip-build-khslhlkr/fontbakery/
$ brew install python; brew install fontforge --HEAD
Warning: python 2.7.14 is already installed
Error: No head is defined for fontforge
20:46 $ brew install python; brew install fontforge 
Warning: python 2.7.14 is already installed
Warning: fontforge 20170731_2 is already installed
20:46 $
```

**Comment by felipesanches** [comment]
fontbakery-0.0.13.tar.gz !!! :-O

**Comment by davelab6** [comment]
I don't know why it picked that version

**Comment by felipesanches** [comment]
Me neither. I'll have to look that up and try to figure out.

**Comment by anthrotype** [comment]
Maybe Dave’s pip is set up look up to a different python package index, maybe a local or corporate one? Just guessing.
Check the pip configuration file at https://pip.pypa.io/en/stable/user_guide/#config-file
And see if it contains any extraneous index-url.
Maybe also try to pass the index url of the official pypi to pip (there’s a command line option).

**Comment by felipesanches** [comment]
I've contacted Marcelo Magalhães trying to schedule a day to meet in person. Then I'll try to debug the installation procedures on his Mac.

**Comment by felipesanches** [comment]
Me and Marcelo worked together today and we fixed the instructions for installing on a Mac.

**Comment by felipesanches** [comment]
Today I got a Macbook. It is not a new one, but it came with a fresh new system. So I decided to validate myself that the install instructions are all good. I took notes of absolutely everything I did. And the install instructions are still bad (or maybe I am a clueless mac user now still needing to learn the basics)

I'll reopen this issue now and then I'll write another message with a detailed report of everything I did up to now (and got that same fontbakery-0.0.13 that was haunting Dave's Mac).

**Comment by felipesanches** [comment]
I started by doing `git clone` which prompted me to install Mac dev tools, to which I pressed the "Install" button.

Then I checked that `python` opens the python 2.7.5 interpreter.

I tried `virtualenv venv` but got command not found. Then I started following our README.md install instructions: https://github.com/googlefonts/fontbakery#install

* `xcode-select --install` game me that same prompt again. I think it was redundant but I clicked the "Install" button again anyway...
* then I installed home-brew as instructed, with: `ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)";`
* I **did not run** `brew install python` because I knew python was already installed on the system (at least 2.7.5 and I did not want to unnecessarily complicate things by having yet an additional version of python installed - maybe that command would install python 3, for instance?)
* next install instruction that I followed was `easy_install pip` and that naturally failed with a permission error.
* I then tried it with sudo: `sudo easy_install pip` and got this error message: `Could not find suitable distribution for Requirement.parse('pip')`
* So I decided to git `brew install python` a chance. After that I got Python 3.6.4 available on my system.
* Then, `pip install fontbakery` gave me that same error that @davelab6 had last year with pip attempting to install a very old version of fontbakery (**version 0.0.13** !!!)
* `pip3 isntall fontbakery` resulted in absolutely the same problem.

**Comment by felipesanches** [comment]
I did try two other things immediately after the actions reported in the previous message:

* `easy-install-3.6 pip` seemed to run properly. But after that, both `pip install fontbakery` and `pip3 install fontbakery` still continued trying to install fb-0.0.13 ...

Then I tried:
* `pip install virtualenv` OK
* `. venv/bin/activate` OK
* Both `pip install font bakery` and `pip3 install fontbakery` still trying 0.0.13 ... :-P

**Comment by felipesanches** [comment]
```
MacBook-Pro-de-Felipe:~ fsanches$ pip -V
pip 9.0.1 from /usr/local/lib/python3.6/site-packages (python 3.6)
```

**Comment by felipesanches** [comment]
The problem is that homebrew installs python 3, while all fontbakery packages are marked as python 2 only.

Creating a virtual environment with the python 2.7 interpreter, I was able to properly install fontbakery 0.3.4:
```
virtualenv -p python2.7 venv2.7
. venv2.7/bin/activate
pip install fontbakery
```

**Comment by felipesanches** [comment]
I saw some recommendations to use a homebrew-installed python because the python interpreter available by default on Mac OS tends to be older than the latest official releases.

The fontbakery packages are not marked as py3 because I am not sure all the code is python 3 ready.

See also #1108

**Comment by felipesanches** [comment]
Also, using a virtualenv is a solution, but I think it is not necessarily the most user-friendly solution as the user will always have to remember to activate the venv before using fontbakery.

**Comment by felipesanches** [comment]
Maybe we should leave the MacOs install instructions based on the default python 2.7.5 for now, until we sort out whatever we still may need to do regarding a python 3 ready fontbakery (issue #1108).

**Comment by graphicore** [comment]
> Also, using a virtualenv is a solution, but I think it is not necessarily the most user-friendly solution as the user will always have to remember to activate the venv before using fontbakery.

Maybe we can somehow build the virtualenv activation into the fontbakery command, or a replacement. Something like:


path/to/bin/fontbakery --> path/to/fontbakery/bin/fontbakery-activate-venv.sh # would be linked
```sh
#! /bin/bash
./path/to/fontbakery/venv/bin/activate
# should now use the fontbakery command of the virtualenv
fontbakery "$@"
```

**Comment by anthrotype** [comment]
2.7.5 is ancient. Do it a favour, leave the system python alone, just brew install python2 to get a recent python2.7 with all the latest bug fixes and security patches, pip preinstalled and up to date, and proceed from there with the setup.

**Comment by felipesanches** [comment]
great! Thanks, @anthrotype !

**Comment by anthrotype** [comment]
i'm actually surprised High Sierra (is that the macOS version you're running) still has 2.7.5...

**Comment by felipesanches** [comment]
yes, I was running Maverick when I did all the commands reported above. Then I noticed I was running an old mac os release, so I installed the upgrade to High Sierra.

Even after upgrading I still got the same versions of python 2.7.5 and 3.6.4

**Comment by felipesanches** [comment]
`brew install python2` now gave me python 2.7.14 ;-)

**Comment by felipesanches** [comment]
but running it, I still get the 2.7.5 interpreter :-(

**Comment by anthrotype** [comment]
yeah, that's a feature. `python` by default is now `python3.6` (YAY!)
https://twitter.com/cosimolupo/status/969714175705534464

you have to call it as `python2`

**Comment by anthrotype** [comment]
or `python2.7`

**Comment by anthrotype** [comment]
> I was running Maverick when I did all the commands reported above. Then I noticed I was running an old mac os release, so I installed the upgrade to High Sierra.
> Even after upgrading I still got the same versions of python 2.7.5

That's very weird. I don't have my mac here with me, but I'm pretty sure `/usr/bin/python --version` returns "2.7.10", not "2.7.5".
Something may have gone wrong with your upgrade from Mavericks (10.9) to High Sierra (10.13).

I googled around and I found several places that confirm High Sierra ships with 2.7.10, not 2.7.5. Here's one link:
https://forums.developer.apple.com/thread/78891
